dist/common/scripts/scylla_util.py

```diff
@@ -117,6 +117,239 @@ def curl(url, headers=None, byte=False, timeout=3, max_retries=5, retry_interval
                 raise
 
 
+class azure_instance:
+    """Describe several aspects of the current AZURE instance"""
+
+    EPHEMERAL = "ephemeral"
+    ROOT = "root"
+    GETTING_STARTED_URL = "http://www.scylladb.com/doc/getting-started-azure/"
+    META_DATA_BASE_URL = "http://169.254.169.254/metadata/instance"
+    ENDPOINT_SNITCH = "SimpleSnitch"
+
+    def __init__(self):
+        self.__type = None
+        self.__cpu = None
+        self.__memoryGB = None
+        self.__nvmeDiskCount = None
+        self.__firstNvmeSize = None
+        self.__osDisks = None
+
+    @staticmethod
+    def is_azure_instance():
+        """Check if it's AZURE instance via DNS lookup to metadata server."""
+        import socket
+        try:
+            addrlist = socket.getaddrinfo('169.254.169.254', 80)
+        except socket.gaierror:
+            return False
+        for res in addrlist:
+            af, socktype, proto, canonname, sa = res
+            if af == socket.AF_INET:
+                addr, port = sa
+                if addr == "169.254.169.254":
+                    return True
+        return False
+
+    def __instance_metadata(self, path, recursive=False):
+        return curl(self.META_DATA_BASE_URL + path + "?api-version=2021-01-01&format=text", headers = { "Metadata": "true" })
+
+    def _non_root_nvmes(self):
+        """get list of nvme disks from os, filter away if one of them is root"""
+        nvme_re = re.compile(r"nvme\d+n\d+$")
+        root_dev_candidates = [x for x in psutil.disk_partitions() if x.mountpoint == "/"]
+        if len(root_dev_candidates) != 1:
+            raise Exception("found more than one disk mounted at root ".format(root_dev_candidates))
```

Comment:
There's no `{}` marker so there's no place where it could print `root_dev_candidates`.

In any case, please use the modern f-string syntax for string formatting:

```raise Exception(f"Found more than one disk mounted as root: {root_dev_candidates}")
```